pkg/kubelet/lifecycle/handlers.go

```diff
@@ -187,6 +187,11 @@ func (a *noNewPrivsAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		return PodAdmitResult{Admit: true}
 	}
 
+	// Always admit for remote runtime.
+	if a.Runtime.Type() == kubetypes.RemoteContainerRuntime {
```

Comment:
@Random-Liu  Seems still not to fix this issue. 
`a.Runtime.Type()` is not a constant value for all of the remote runtime,  It's get from remote API, ref codes:
https://github.com/kubernetes/kubernetes/blob/65da3ce2466d88f50e138051d26ded166348e55f/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L152-L167

Reply:
In my test, The error print contains the following content:
```
Cannot enforce NoNewPrivs: containerd runtime not supported
```
That to say, the value of `a.Runtime.Type()` is `containerd` for cri-containerd but not `remote`.

Is my analysis correct? : )

Reply:
@miaoyq You are right, `Runtime.Type()` is the actually runtime name instead of `remote`.

Reply:
Filed #51404 for fixing it.

Reply:
Thanks @feiskyer

Reply:
@miaoyq Thanks for catching this. This is really confusing, we sometimes use the runtime names from flag to do admission check, which are "remote"/"docker". Sometimes using the runtime names got through CRI, which are "docker"/"containerd"/"cri-o".